cmd/flightctl-api/main.go

```diff
@@ -187,32 +199,48 @@ func main() {
 
 	orgResolver, err := resolvers.BuildResolver(buildResolverOpts)
 	if err != nil {
-		log.Fatalf("failed to build organization resolver: %v", err)
+		return err
 	}
 
 	agentServer, err := agentserver.New(ctx, log, cfg, store, ca, agentListener, provider, agentTlsConfig, orgResolver)
 	if err != nil {
-		log.Fatalf("initializing agent server: %v", err)
+		return err
 	}
 
+	// Create API server listener and server
+	listener, err := middleware.NewTLSListener(cfg.Service.Address, tlsConfig)
+	if err != nil {
+		return err
+	}
+	// we pass the grpc server for now, to let the console sessions to establish a connection in grpc
+	apiServer := apiserver.New(log, cfg, store, ca, listener, provider, agentServer.GetGRPCServer(), orgResolver)
+
+	// Channel to collect server errors
+	errCh := make(chan error, 3)
+
+	// WaitGroup to track server goroutines
+	var serverWg sync.WaitGroup
```

Comment:
simple is clean I totally get your intentions but we only have one wait.Group

```suggestion
	var wg sync.WaitGroup
```